### PR TITLE
UPDATE: Basic Inbox Functionality Finished

### PR DESCRIPTION
**Pull request by lewisning**
# Description

Currently finished the display of following requests on the inbox page, however, it still has other displays needed to do such as posts, likes.

Alisha finished the back end of the inbox, Chen finished the front end and database connection, I assisted Chen to connect the database.

## Change 1: UI
Created two individual inbox pages for messages and requests. Currently, we only design to display all requests on the message page.
截屏2022-03-03 13 39 32

Use /inbox/messages to access the message page.
Use /inbox/requests to access the request page

## Change 2: The notification dot
Case 1: When there is no data in the inbox database, the red dot will not be displayed and the message: "There is no messages in inbox right now." will be displayed on the message page.
截屏2022-03-03 13 47 53
截屏2022-03-03 13 48 18

Case 2: When the inbox database contains data, the red dot will be displayed on the up-right corner of the specific item in the dropdown list.
截屏2022-03-03 13 39 32

When there is a request message in the inbox, the decline and accept buttons will be displayed on the right of the request summary.
截屏2022-03-03 13 46 57

## Designed for now:
1. Both items in the dropdown list will show the red dot if there are some data in the inbox database.
2. The inbox categorization will be done in the next part
3. Requests page only contains received follow requests
4. Massage page contains posts, likes and all other data from follow requests
5. Requests page is not used for now, all data from the inbox database will be displayed on the message page

### CODE REVIEW

File: home/templates/messages.html [hunk 1]
```diff
@@ -14,16 +14,19 @@
...
+                <div class="rows">
+                    <div class="column">
+                        <p>{{ content.summary }}</p>
+                    <span class="actions">
+                        <button class="confirm">Confirm</button>
+                        <button class="reject">Decline</button>
+                    </span>
+                    </div>
+                </div>
+{#            {% endfor %}#}
```
Comment (LefanTan) [inline home/templates/messages.html:28]: what are these {# ?

File: home/views.py [hunk 2]
```diff
@@ -87,21 +92,53 @@ def post_create(request):
...
+
+    try:
+        # page_number = request.GET.get('page') or 1
+        # size = request.GET.get('size') or 5
+
+        item = InboxObject.objects.get(object=author)
+        model = item.content_type.model_class()
+
+        # paginator = Paginator(item, size)
+        # requests = paginator.get_page(page_number).object_list
```
Comment (LefanTan) [inline home/views.py:128]: comments shouldn't be added to PR. Removed if not needed, if needed, add another comment to explain why

**Review by LefanTan — CHANGES_REQUESTED** [review]
I have more questions:
1. Out of curiosity, are friends adding implemented? If not, does that mean anyone can share a post to anyone right? 
2. Also, private post can be shared?
3. In the screenshots provided, why does /inbox/messages show follow request? It should be /inbox/request right? 
4. Can you provide another screenshot as to what /inbox/messages with populated data look like?

File: home/templates/messages.html (hunk 1 above)
Comment (lewisning) [reply home/templates/messages.html:28]: these are checking the inbox items and displaying them. the for loop hasn't been finished

File: home/views.py (hunk 2 above)
Comment (lewisning) [reply home/views.py:128]: okay

**Review by LefanTan — APPROVED** [review]

**Pull request merged**